design_documents/parametrizing_tests_with_fixed_values.md

```diff
@@ -0,0 +1,106 @@
+# Parametrizing Tests With Fixed Values
+
+## Context
+
+Various test runners for other programming languages provide ways of parametrizing test cases with fixed values.
+This way, multiple test cases can be executed for the same test source code.
+
+## Goal
+
+Propose a way of adding support for parametrized tests to Starknet Foundry's `snforge`.
+
+## Proposed Solution
+
+Introduce three new attributes: `parametrize`, `case` and.
+
+- `parametrize` - defines a parametrized test.
+- `case` - defines a specific test case
+
+### Parametrization With Simple Arguments
+
+```cairo
+#[parametrize]
```

Comment:
Would this be necessary?

Reply:
Probably not, we can remove it. Perhaps having this attribute could simplify some handling in the actual implementation, though it's hard to say.

Reply:
how about 

```
#[parametrize(
    case(...),
    case(...),
]
```

would that be possible?

Reply:
Imo this is too verbose. We should drop `test` and `parametrize`

`#[test_case(1,2,)]` is sufficient an consistent with rust. 

The main goal of this syntax is to write succinct tests, so it is better to keep the syntax succinct.

Reply:
Can be done I think

Reply:
@MaksymilianDemitraszek +10